### PR TITLE
Add .pc file generation to Makefile

### LINKED ISSUE #2231: pkg-config support (tracking bug)
We should do the following:

 - [x] Use `pkg-config` to check dependencies (protobuf, openssl) when possible in the Makefile. (#2233)
 - [x] Install `.pc` files for `gpr` and `grpc` libraries. (#2253)
 - [ ] Use `pkg-config` for the `grpc` dependency in wrapped language build scripts.
 - [ ] Modify the tests

Then we should change the homebrew script to only install the core and C++ libraries, because homebrew interacts well with `pkg-config` and this would allow us to use standard language tools `npm`, `pip`, `gem`, etc. to install the wrapped language packages.

This should allow us to more easily fix #2220, grpc/grpc-common#233, grpc/homebrew-grpc#31, and grpc/homebrew-grpc#32 (and possibly others).

### LINKED ISSUE #2253: Generate and install pkg-config files from Makefile
This will allow other libraries to depend on gRPC using `pkg-config`, including the wrapping libraries. This is part of #2231.

### PR DESCRIPTION
**Pull request by murgatroid99**
This fixes #2253

### CODE REVIEW

File: templates/Makefile.template [hunk 1]
```diff
@@ -347,6 +347,31 @@ HOST_LDLIBS = $(LDLIBS)
...
+Cflags: -I${'\$${includedir}'} $(PC_CFLAGS)\n\
+Requires.private: $(PC_REQUIRES_PRIVATE)\n\
+Libs: -L${'\$${libdir}'}\n\
+Libs.private: $(PC_LIBS_PRIVATE)
+
+# gpr .pc file
+PC_NAME = Google Public Runtime
+PC_DESCRIPTION = Google Public Runtime
+PC_CFLAGS =
+PC_REQUIRES_PRIVATE =
```
Comment (nicolasnoble) [inline templates/Makefile.template]: Wasn't this supposed to read -pthreads and -lpthreads ?

Comment (ctiller) [inline templates/Makefile.template]: I think P was for portable.

Comment (murgatroid99) [inline templates/Makefile.template]: Yes, it was. And now it does.

File: Makefile [hunk 2]
```diff
@@ -333,6 +333,19 @@ HOST_LDLIBS = $(LDLIBS)
 
 HAS_PKG_CONFIG = $(shell command -v pkg-config >/dev/null 2>&1 && echo true || echo false)
 
+PC_TEMPLATE = prefix=$(prefix)\nexec_prefix=\$${prefix}\nincludedir=\$${prefix}/include\nlibdir=\$${exec_prefix}/lib\n\nName: $(PC_NAME)\nDescription: $(PC_DESCRIPTION)\nVersion: $(VERSION)\nCflags: -I\$${includedir} $(PC_CFLAGS)\nRequires.private: $(PC_REQUIRES_PRIVATE)\nLibs: -L\$${libdir}\nLibs.private: $(PC_LIBS_PRIVATE)
+
+# gpr .pc file
+PC_NAME = Google Public Runtime
+PC_DESCRIPTION = Google Public Runtime
```
Comment (nicolasnoble) [inline Makefile]: And @ctiller is right, my neurons fizzled. GPR stands for Google Portable Runtime.

Comment (murgatroid99) [inline Makefile]: Changed it to "gRPC Portable Runtime" as we discussed.

**Pull request merged**